Framework/PythonInterface/plugins/algorithms/PowderReduceP2D.py

```diff
@@ -0,0 +1,723 @@
+        grp4 = 'FilterBadPulses'
+        self.setPropertyGroup('LowerCutoff', grp4)
+        # Input for RemovePromptPulse
+        self.declareProperty(
+            'Width',
+            150,
+            direction=Direction.Input,
+            doc=
+            'The width of the time of flight (in microseconds) to remove from the data.'
+        )
+        self.copyProperties('RemovePromptPulse', ['Frequency'])
+        grp5 = 'RemovePromptPulse'
+        self.setPropertyGroup('Width', grp5)
+        self.setPropertyGroup('Frequency', grp5)
+        # Input for LoadDiffCal
+        self.declareProperty(
+            'WorkspaceName',
+            'POWTEX',
+            direction=Direction.Input,
+            doc=
+            'The base of the output workspace names. Names will have _group, _cal, _mask appended to them.'
+        )
+        self.copyProperties('LoadDiffCal', [
+            'InstrumentName', 'InstrumentFilename', 'MakeGroupingWorkspace',
+            'MakeCalWorkspace', 'MakeMaskWorkspace', 'TofMin', 'TofMax',
+            'FixConversionIssues'
+        ])
+        grp6 = 'LoadDiffCal'
+        self.setPropertyGroup('WorkspaceName', grp6)
+        self.setPropertyGroup('InstrumentName', grp6)
+        self.setPropertyGroup('InstrumentFilename', grp6)
+        self.setPropertyGroup('MakeGroupingWorkspace', grp6)
+        self.setPropertyGroup('MakeCalWorkspace', grp6)
+        self.setPropertyGroup('MakeMaskWorkspace', grp6)
+        self.setPropertyGroup('TofMin', grp6)
+        self.setPropertyGroup('TofMax', grp6)
+        self.setPropertyGroup('FixConversionIssues', grp6)
+        # Input for MaskDetectors
+        self.declareProperty(
+            'MaskedWorkspace',
+            self.getProperty('WorkspaceName').value + '_mask',
+            direction=Direction.Input,
+            doc=
+            'If given but not as a SpecialWorkspace2D, the masking from this workspace will be copied. If given as a '
+            'SpecialWorkspace2D, the masking is read from its Y values.'
+        )
+        self.copyProperties('MaskDetectors', [
+            'SpectraList', 'DetectorList', 'WorkspaceIndexList',
+            'ForceInstrumentMasking', 'StartWorkspaceIndex',
+            'EndWorkspaceIndex', 'ComponentList'
+        ])
+        grp7 = 'MaskDetectors'
+        self.setPropertyGroup('MaskedWorkspace', grp7)
+        self.setPropertyGroup('SpectraList', grp7)
+        self.setPropertyGroup('DetectorList', grp7)
+        self.setPropertyGroup('WorkspaceIndexList', grp7)
+        self.setPropertyGroup('ForceInstrumentMasking', grp7)
+        self.setPropertyGroup('StartWorkspaceIndex', grp7)
+        self.setPropertyGroup('EndWorkspaceIndex', grp7)
+        self.setPropertyGroup('ComponentList', grp7)
+        # Input for AlignDetectors
+        #self.copyProperties('AlignDetectors', ['CalibrationWorkspace', 'OffsetsWorkspace'])
+        #grp8 = 'AlignDetectors'
```

Comment:
My plan in the beginning was to allow the user of the workflow algorithm to potentially change the input of all functions called. I had some problems with that particular function and commented that part out because of it. I might get back to it later, but at the moment have no need for that part of the code. Its basically a reminder where to add it at a later time. 
I could just remove it if thats no good reason for leaving it of course.